Testes feitos

Djeymisson opened the pull request.

Commit:
Adicionado um teste de controlador na página de Coleta

Commit:
Cenario 'there is no residues to collect' criado'

Commit:
criado cenario media negativa

Commit:
Adicionado teste sobre o controlador de Historico de coleta

Commit:
Adicionando um teste de GUI à página de Registro de Gerador

Commit:
cenario de gui criado

Commit:
criado teste de controlador sobre nome de coleta ser vazio

NatalicioCavalcante committed:
Novo Teste de Controlador

Commit:
Teste de controlador incompleto.

Commit:
teste de controlador adicionado para a classe ResidueGenerator

Commit:
Adicionado teste de GUI - Alysson

Commit:
corrigido o teste sobre a classe ResidueGenerator

Commit:
efetuado merge com remote de Eberson

Commit:
Adicionado teste de GUI(inserindo uma 'string' no volume de criador de coleta que deveria ser 'int') - Alysson

Commit:
efetuado merge com remote de Alysson

Commit:
Merge remote-tracking branch 'alyssonmanso/master'

Commit:
corrigido erros decorridos do merge automático do git

NatalicioCavalcante committed:
Alterações no Teste de Controlador

NatalicioCavalcante committed:
Teste de Controlador, Impedir que Seja Adicionado 2 Geradores de Residuos com o Mesmo CNPJ, Deveria Estar Funcionando

NatalicioCavalcante committed:
Teste De GUI - Testar se Não é Adicionada a Coleta Com Nome Vazio

Djeymisson committed:
Efetuado merge com repositório de Natalicio.

Commit:
Corrigido conflitos decorrente do merge

Commit:
funcionalidade criar user adicionada

Commit:
Merge remote-tracking branch 'Djeymisson/master'

Commit:
adicionada funcionalidade de listar

Commit:
merge para completar os testes

Commit:
alterações em alguns testes (feitos por Natalicio) e criação da funcionalidade de editar user.

Commit:
Efetuado merge com o repositorio de Eberson

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -82,4 +82,9 @@ Feature: Gerar historico de coleta
     And existe uma coleta na listagem
     And seleciono esta coleta
     When aperto o botao Delete
-    Then estou na pagina de listagem e esta coleta nao consta mais
```
evitar esse tipo de alteração desnecessária

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -82,4 +82,9 @@ Feature: Gerar historico de coleta
+  Scenario: adicionar coleta com campo nome vazio
+  	Given eu estou na pagina de adicionar coleta
+  	When eu adiciono uma nova coleta com o nome ""
+  	Then a coleta nao e adicionada
```
qual coleta não é adicionada? tem que especificar com o uso de parâmetro

rcaa commented on test/cucumber/steps/HistoricoDeColetasStep.groovy:
```diff
@@ -142,4 +142,22 @@ When(~'^aperto o botao Delete$') { ->
+	at HistoricoPage
+}
+
+def nomeParametro
```
não usar atributos de classe

rcaa commented on test/cucumber/steps/RegisterGeneratorSteps.groovy:
```diff
@@ -96,27 +96,39 @@ When(~'I fill the generator details with some fields left incomplete$'){->
     page.fillGeneratorDetailsIncomplete(generator)
 
 }
+def newGenerator
```
não fazer definições de classe

rcaa commented on test/cucumber/steps/RegisterGeneratorSteps.groovy:
```diff
@@ -96,27 +96,39 @@ When(~'I fill the generator details with some fields left incomplete$'){->
-                            averageDailyMeals: 0]
-		newGenerator.save()
+
+		newGenerator = new ResidueGenerator()
```
usar o "params" e chamar o save de controlador, como discutimos durante a aula

rcaa commented on test/cucumber/steps/RegisterGeneratorSteps.groovy:
```diff
@@ -96,27 +96,39 @@ When(~'I fill the generator details with some fields left incomplete$'){->
-                            averageMonthlyMeals: 0,
-                            averageDailyMeals: 0]
-		newGenerator.save()
+		newGenerator2 = new ResidueGenerator()
```
mesmo comentário anterior pra esse passo

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -17,12 +17,6 @@ Feature: Gerar historico de coleta
   	Given nao existe uma coleta com nome ""
   	When eu crio uma nova coleta com nome "" e data "08/05/2015"
   	Then a coleta com nome "" nao sera criada
-  	
```
quem fez esse commit esqueceu de configurar usuário a commitar, como tá no slide da aula. Aqui aparece "PCUAG". Por favor identifiquem com comentários quem commitou o que.

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -17,12 +17,6 @@ Feature: Gerar historico de coleta
-  	
-  Scenario: adicionar coleta com dados invalidos em volume
-    Given estou na pagina de adicionar coleta
-    When preencho o campo volume com valor "dois"
```
Qual o propósito deste commit? "Testes feitos" é a mensagem, mas só tem remoção de código aqui.

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -8,6 +8,12 @@ Feature: Gerar historico de coleta
+  	
+  Scenario: adicionar coleta com dados invalidos em volume
+    Given estou na pagina de adicionar coleta
+    When preencho o campo volume com valor "dois"
```
depois de preencher o campo volume já vê a mensagem de erro? Não precisa clicar em adicionar? Caso precise, tem que especificar isso em um passo adicional.

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -8,6 +8,12 @@ Feature: Gerar historico de coleta
+  Scenario: adicionar coleta com dados invalidos em volume
+    Given estou na pagina de adicionar coleta
+    When preencho o campo volume com valor "dois"
+    Then eu vejo uma menssagem de erro
```
descrição do passo muito vaga. Qual mensagem de erro? Colocar um parâmetro para identificar a mensagem como sendo proveniente do erro ao adicionar a coleta com nome inválido

rcaa commented on test/functional/pages/HistoricoPage.groovy:
```diff
@@ -20,6 +20,26 @@ class HistoricoPage extends Page{
+	}
+
+	def boolean hasInvalidMessage(){
+		
```
olha a identação despadronizada. Tem muito mais espaço que nos demais trechos de código.

rcaa commented on test/cucumber/RegisterGenerator.feature:
```diff
@@ -44,4 +44,4 @@ Feature: register a residue generator
   	Given doesn't exist a residue generator with address "" stored in the system
   	When I create a residue generator with address ""
-  	Then the new residue is not stored
+  	Then the new residue with address "" is not stored
```
o endereço não é null. O endereço é uma string vazia. Corrigir descrição do cenário.

rcaa commented on test/cucumber/HistoricoDeColeta.feature:
```diff
@@ -2,6 +2,12 @@ Feature: Gerar historico de coleta
+ Scenario: adicionar coleta com dados invalidos em volume
+ Given estou na pagina de adicionar coleta
+ When preencho o campo volume com valor "dois"
+ Then eu vejo uma menssagem de erro
```
mensagem tá escrito errado

rcaa commented on test/cucumber/EditResidueGenerator.feature:
```diff
@@ -50,4 +50,9 @@ Feature: Edit the residue generator registration data
     And fill all the required fields correctly
     And submit my changes
     Then I should see a message indicating that an error occurred
+    
```
se tá incompleto, não era pra ter sido commitado

rcaa commented on test/cucumber/CreateHarvestSolicitation.feature:
```diff
@@ -7,10 +7,11 @@ Feature: Create a harvest solicitation
     Then  the system should store the solicitation
-
+    
+@ignore
```
por que este cenário foi alterado e está sendo ignorado?

rcaa commented on test/cucumber/steps/HistoricoDeColetasStep.groovy:
```diff
@@ -21,6 +21,20 @@ And (~'^clico em adicionar coleta do dia$'){ ->
+When(~/^eu crio uma nova coleta com nome "(.*?)" e data "(.*?)"$/) {String nome, @Format("dd/MM/yyyy") Date dia ->
+	coleta = new Coleta()
+    coleta.name = nome
+    coleta.data = dia
```
código não corresponde a descrição do cenário. a coleta não é salva pelo controlador

rcaa commented on test/cucumber/EditResidueGenerator.feature:
```diff
@@ -43,4 +43,11 @@ Feature: Edit the residue generator registration data
+  Scenario: Residue generator registration data edit fail
+    Given I am at the residue generator edit page
+    When I fill the name field with ""
+    And fill all the required fields correctly
```
preenche com o que? a descrição do passo está vaga

rcaa commented on test/cucumber/EditResidueGenerator.feature:
```diff
@@ -43,4 +43,11 @@ Feature: Edit the residue generator registration data
+    Given I am at the residue generator edit page
+    When I fill the name field with ""
+    And fill all the required fields correctly
+    And submit my changes
```
o que significa "my changes"?

rcaa commented on test/cucumber/RegisterGenerator.feature:
```diff
@@ -39,4 +44,4 @@ Feature: register a residue generator
     When I fill the generator details with some fields left incomplete
     And I register the new generator
-    Then I should see an error message
+    Then I should see an error message
```
qual mensagem de erro?

rcaa commented on test/functional/steps/GeneratorTestDataAndOperations.groovy:
```diff
@@ -71,6 +71,20 @@ class GeneratorTestDataAndOperations{
+	static public void createGenerator(String name, int average){
+		def cont = new ResidueGeneratorController()
+		def newGenerator = [nome: name,
+							type: "Restaurante",
```
corrigir identação do código

rcaa commented on test/cucumber/steps/HistoricoDeColetasStep.groovy:
```diff
@@ -142,4 +142,19 @@ When(~'^aperto o botao Delete$') { ->
+}
+
+Given(~/^o sistema nao possui uma coleta com nome "(.*?)" e data "(.*?)"$/) { String arg1, String arg2 ->
+	// Write code here that turns the phrase above into concrete actions
```
nunca deve-se commitar algo que não está completo

NatalicioCavalcante committed:
Alterações no teste de controlador, como feito em sala..
O teste é para falhar, pois são criados 2 objetos com o mesmo cnpj, o que não deveria ser permitido.

Djeymisson reviewed (commented):
Djeymisson - "PCUAG", e "uag-pc"
Alysson Manso - 'Usuario' e 'alyssonmanso'
Eberson Santos - "pc-uag"

Djeymisson committed:
Efetuado merge com o repositorio de Natalicio, para adicionar o último teste dele.

Djeymisson committed:
Corrigido alguns erros decorrentes do merge.

EbersonSantos committed:
correção de erro: data mínima em uma coleta

alyssonmanso committed:
Alysson - Merge com as funcionalides de criar e editar feitas por Eberson e Djeymisson

alyssonmanso committed:
Alysson - Corrigido erro de criar coleta com volume negativo ou vazio

Djeymisson committed:
Corrigido alguns erros envolvendo o cadastro de cnpj no ResidueGenerator

Djeymisson committed:
Merge remote-tracking branch 'AlyssonManso/master'

Djeymisson committed:
Adicionada a UML de classe, e efetuado o merge com o repositorio de Alysson, e Eberson.

alyssonmanso committed:
Alysson - Adicionado metodo toString() na classe HarvestCompany.groovy

Djeymisson committed:
Correcao no diagrama de classes.

EbersonSantos committed:
refatoracao - adicionado metodo toString() a classe Coleta

Djeymisson committed:
Corrigido alguns erros apontados pelo CodeNarc

NatalicioCavalcante committed:
Alterações no teste de Controlador, tal como falado em sala.. Agora o teste tenta garantir que seja possível criar 2 objetos com o mesmo CNPJ...

NatalicioCavalcante committed:
Função deletar da nova funcionalidade do sistema...

NatalicioCavalcante committed:
Resolvi o erro do teste da feature HomePage, onde o mesmo agora consegue completar o teste de acesso ao repositorio ReeS - Paulo Borba no Git Hub...

Djeymisson committed:
Feito merge com o repositorio de Natalicio.

rcaa reviewed (commented):
revisão

rcaa commented on grails-app/conf/BootStrap.groovy:
```diff
@@ -41,10 +41,34 @@ class BootStrap {
                 harvestSolicitation: null
```
esse commit deveria ter sido quebrado em dois:
- sua funcionalidade
- testes de natalicio

rcaa commented on grails-app/controllers/user/UserController.groovy:
```diff
@@ -0,0 +1,39 @@
+package user
+
```
a mensagem do commit deve ser mais explicativa. Por exemplo:
"Adicionando a funcionalidade de deletar um usuario"

rcaa commented on test/functional/pages/GithubPage.groovy:
```diff
@@ -4,7 +4,7 @@ import geb.Page
 
 class GithubPage extends Page {
 
-    def titulo = "pauloborba/ResS � GitHub"
```
A correção tinha que ser em alguma funcionalidade do sistema identificada por algum teste ou pelo uso da aplicação. Essa correção aqui corresponde a correção de um teste.

rcaa commented on test/functional/pages/GithubPage.groovy:
```diff
@@ -4,7 +4,7 @@ import geb.Page
 class GithubPage extends Page {
 
-    def titulo = "pauloborba/ResS � GitHub"
+    def titulo = "GitHub - pauloborba/ResS: residue management system"
```
O erro a ser corrigido deveria ter sido referente a alguma funcionalidade do sistema e não a um teste. O correto seria identificar algum problema rodando testes ou rodando a aplicação.

rcaa commented on grails-app/controllers/user/UserController.groovy:
```diff
@@ -0,0 +1,39 @@
+            form multipartForm {
+                flash.message = message(code: 'default.not.found.message', args: [message(code: 'user.label', default: 'User'), params.id])
+                redirect action: "index", method: "GET"
+            }
```
Como o usuário acessaria essa nova funcionalidade? Faltou a parte de GUI.

Djeymisson committed:
Corrigido um erro decorrente do merge.

EbersonSantos committed:
refatoracao - extracao de metodo para criar geradores de residuo

Djeymisson committed:
Efetuado refatoração, extraindo método em comum.

alyssonmanso committed:
Alysson - Refatoração: Extração de Método na classe HistoricoTestDataAndOperations.groovy metodos de createHistorico

Djeymisson committed:
Feito merge com Eberson

Djeymisson committed:
Efetuado merge com o repositorio de Alysson.

NatalicioCavalcante committed:
Remoção de Classe não utilizada - Sugerido pelo Codenarc

NatalicioCavalcante committed:
Algoritmo de Substituição - Refatoração
Fiz uma mudança do método createGeneratorNomeCnpj da classe GeneratorTestDataAndOperations, reduzindo o seu tamanho e tornando mais simples, tendo o mesmo resultado...

Djeymisson committed:
Efetuado merge com o repositorio de Natalicio.
